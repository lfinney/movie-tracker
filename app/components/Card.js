import React from 'react';
import PropTypes from 'prop-types';

const dateFormat = require('dateformat');
// const now = new Date();


const Card = ({ movieData, addToFavorites, userId, userFavArray, removeFav, type }) => {
  const poster = `https://image.tmdb.org/t/p/w500/${movieData.poster_path}`;
  const backdrop = `https://image.tmdb.org/t/p/w500/${movieData.backdrop_path}`;
  const favoriteId = userFavArray.find( fav => fav.title === movieData.title).movie_id
	  console.log(favoriteId)
	  
  return (
    <div className={ userFavArray.find( movie => movie.movie_id === movieData.movie_id) ? "card-div favorite" : "card-div"}>
      <img src={movieData.poster_path} />
      <img src="../styles/assets/fav-icon.svg" className="fav-icon"/>
      <div className="card-text">
        <p className="title">{movieData.title}</p>
        <p className="date">{dateFormat(movieData.release_date, "longDate")}</p>
        <p>Avg. User Rating: {movieData.vote_average}</p>
        <p>{movieData.overview}</p>
        <button onClick={() => addToFavorites(movieData, userId, userFavArray)}>Favorite</button>
	{ type === "favs" && <button onClick={() => removeFav(userId, favoriteId)}>rm fav</button> }
      </div>
    </div>
  );
};

Card.propTypes = {
  movieData: PropTypes.object,
  addToFavorites: PropTypes.func,
  userId: PropTypes.number
};

export default Card;
