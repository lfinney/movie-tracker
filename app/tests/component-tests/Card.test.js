import React from 'react';
import { shallow } from 'enzyme';
import Card from '../../components/Card';

describe('Card', () => {

  it('should render component', () => {
    const mockMovData = {
       movie_id: 346364,
       overview: "In a small town in Maine, seven children known as The Losers Club come face to face with life problems, bullies and a monster that takes the shape of a clown called Pennywise.",
       poster_path: "https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg",
       release_date: "2017-09-05",
       title: "It",
       vote_average: 7.4
   }
    const mockUserFavArray = [
{
       movie_id: 346364,
       id: 12,
       user_id: 2,
       overview: "In a small town in Maine, seven children known as The Losers Club come face to face with life problems, bullies and a monster that takes the shape of a clown called Pennywise.",
       poster_path: "https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg",
       release_date: "2017-09-05",
       title: "It",
       vote_average: 7.4
   }
    ]
	  
    const wrapper = shallow(<Card
		   	movieData={mockMovData}
			addToFavorites={jest.fn()}
			userId='2'
			userFavArray={mockUserFavArray}
			removeFav={jest.fn()}
			type='home'
			key='1'
		    />);

    expect(wrapper).toMatchSnapshot();
  });
});
