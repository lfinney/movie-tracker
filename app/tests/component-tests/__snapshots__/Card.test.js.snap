// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addToFavorites={[Function]}
    movieData={
      Object {
        "movie_id": 346364,
        "overview": "In a small town in...",
        "poster_path": "https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg",
        "release_date": "2017-09-05",
        "title": "It",
        "vote_average": 7.4,
      }
    }
    removeFav={[Function]}
    type="home"
    userFavArray={
      Array [
        Object {
          "id": 12,
          "movie_id": 346364,
          "overview": "In a small town in",
          "poster_path": "https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg",
          "release_date": "2017-09-05",
          "title": "It",
          "user_id": 2,
          "vote_average": 7.4,
        },
      ]
    }
    userId="2"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="poster of It"
          src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
        />,
        <img
          alt="fav-icon"
          className="fav-icon"
          src="../styles/assets/fav-icon.svg"
        />,
        <div
          className="card-text"
        >
          <p
            className="title"
          >
            It
          </p>
          <p
            className="date"
          >
            September 4, 2017
          </p>
          <p>
            Avg. User Rating: 
            7.4
          </p>
          <p>
            In a small town in...
          </p>
          <button
            onClick={[Function]}
          >
            Favorite
          </button>
        </div>,
      ],
      "className": "card-div favorite",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "alt": "poster of It",
          "src": "https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "alt": "fav-icon",
          "className": "fav-icon",
          "src": "../styles/assets/fav-icon.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="title"
            >
              It
            </p>,
            <p
              className="date"
            >
              September 4, 2017
            </p>,
            <p>
              Avg. User Rating: 
              7.4
            </p>,
            <p>
              In a small town in...
            </p>,
            <button
              onClick={[Function]}
            >
              Favorite
            </button>,
            false,
          ],
          "className": "card-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "It",
              "className": "title",
            },
            "ref": null,
            "rendered": "It",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "September 4, 2017",
              "className": "date",
            },
            "ref": null,
            "rendered": "September 4, 2017",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Avg. User Rating: ",
                7.4,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Avg. User Rating: ",
              7.4,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "In a small town in...",
            },
            "ref": null,
            "rendered": "In a small town in...",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Favorite",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Favorite",
            "type": "button",
          },
          false,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="poster of It"
            src="https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
          />,
          <img
            alt="fav-icon"
            className="fav-icon"
            src="../styles/assets/fav-icon.svg"
          />,
          <div
            className="card-text"
          >
            <p
              className="title"
            >
              It
            </p>
            <p
              className="date"
            >
              September 4, 2017
            </p>
            <p>
              Avg. User Rating: 
              7.4
            </p>
            <p>
              In a small town in...
            </p>
            <button
              onClick={[Function]}
            >
              Favorite
            </button>
          </div>,
        ],
        "className": "card-div favorite",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "alt": "poster of It",
            "src": "https://image.tmdb.org/t/p/w500//9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "alt": "fav-icon",
            "className": "fav-icon",
            "src": "../styles/assets/fav-icon.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="title"
              >
                It
              </p>,
              <p
                className="date"
              >
                September 4, 2017
              </p>,
              <p>
                Avg. User Rating: 
                7.4
              </p>,
              <p>
                In a small town in...
              </p>,
              <button
                onClick={[Function]}
              >
                Favorite
              </button>,
              false,
            ],
            "className": "card-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "It",
                "className": "title",
              },
              "ref": null,
              "rendered": "It",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "September 4, 2017",
                "className": "date",
              },
              "ref": null,
              "rendered": "September 4, 2017",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Avg. User Rating: ",
                  7.4,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Avg. User Rating: ",
                7.4,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "In a small town in...",
              },
              "ref": null,
              "rendered": "In a small town in...",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Favorite",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Favorite",
              "type": "button",
            },
            false,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
